src/main/haskell/kore/src/Kore/Builtin/Builtin.hs

```diff
@@ -349,69 +346,112 @@ verifySymbolArguments
-        _ -> return ()
+    :: MonadVerify m
+    => DomainValueVerifiers m child
+    -> (Id Object -> m HookedSortDescription)
+    -> DomainValue Object Domain.Builtin child
+    -> m (DomainValue Object Domain.Builtin child)
+verifyDomainValue
+    verifiers
+    findSort
+    dv@DomainValue { domainValueSort = (SortActualSort _) }
+  = do mHookSort <- lookupHookSort findSort $ domainValueSort dv
+       case mHookSort of
+            Nothing -> defaultDomainValueVerifier dv
+            Just hookSort -> verifyHookedSortDomainValue verifiers hookSort dv
+verifyDomainValue
+    _verifiers
+    _findSort
+    dv@DomainValue { domainValueSort = (SortVariableSort _) }
+  = defaultDomainValueVerifier dv
+
+{- | In the case of a hooked sort, lookup the specific verifier for the hook
+  sort and attempt to encode the domain value. -}
+verifyHookedSortDomainValue
+    :: MonadVerify m
+    => DomainValueVerifiers m child
+    -> Text
+    -> DomainValue Object Domain.Builtin child
+    -> m (DomainValue Object Domain.Builtin child)
+verifyHookedSortDomainValue verifiers hookSort dv = do
+    verifier <- case HashMap.lookup hookSort verifiers of
+                    Nothing -> return defaultDomainValueVerifier
+                    Just verifier -> return verifier
+    validDomainValue <-
+        Kore.Error.withContext
+            ("Verifying builtin sort '" ++ Text.unpack hookSort ++"'")
+            (verifier dv)
+    return validDomainValue
+
+{- | In the case of either variable, unhooked, or non builtin sorts
+  confirm the domain value argument is a string literal pattern.
+-}
+defaultDomainValueVerifier
+    :: MonadVerify m
+    => DomainValueVerifier m child
+defaultDomainValueVerifier
+    dv@DomainValue
+    { domainValueChild = (Domain.BuiltinPattern (StringLiteral_ _))}
+  = return dv
+defaultDomainValueVerifier _
+  = Kore.Error.koreFail "Domain value argument must be a literal string."
+
+-- | Construct a DomainValueVerifier for an encodable sort.
+makeEncodedDomainValueVerifier
+    :: MonadVerify m
+    => Text
+    -- ^ Builtin sort identifier
+    -> ( DomainValue Object Domain.Builtin child -> m (Domain.Builtin child) )
+    -- ^ encoding function for the builtin sort
+    -> DomainValueVerifier m child
+makeEncodedDomainValueVerifier
+    _builtinSort
+    encodeSort
+    domainVal
+  = do dvChild' <- encodeSort domainVal
+       return $ domainVal {domainValueChild = dvChild'}
+
+-- | Construct a 'DomainValueVerifier@ for a sort with no builtin encoding
+makeNonEncodedDomainValueVerifier
+    :: MonadVerify m
+    => Text
+    -> ( DomainValue Object Domain.Builtin child -> m () )
+    -> DomainValueVerifier m child
+makeNonEncodedDomainValueVerifier _builtinName verifyNoEncode domainValue
+  = verifyNoEncode domainValue >> return domainValue
+
+
+{- | Run a parser in a domain value pattern and construct the builtin
+  representation of the value.
+
+  An error is thrown if the string is not a literal string or a previously
```

Comment:
Why does this need to handle previously-encoded domain values?

(Later I would like unencoded and encoded domain values to have different types.)